src/Controller/HearingBriefingController.php

```diff
@@ -0,0 +1,252 @@
+                $relation = new CaseDocumentRelation();
+                $relation->setCase($case);
+                $relation->setDocument($document);
+
+                $this->entityManager->persist($relation);
+                $this->entityManager->persist($document);
+                $this->entityManager->persist($briefingRecipient);
+
+                $briefing->addHearingBriefingRecipient($briefingRecipient);
+            }
+
+            // TODO: Skal alle udsendte høringsskrivelser medsendes?
+            foreach ($hearingPost->getHearingRecipients() as $hearingRecipient) {
+                $briefing->addAttachment($hearingRecipient->getDocument());
+            }
+
+            $this->entityManager->persist($briefing);
+            $this->entityManager->flush();
+
+            $this->addFlash('success', new TranslatableMessage('Hearing briefing updated', [], 'case'));
+
+            return $this->redirectToRoute('case_hearing_index', ['id' => $case->getId(), 'hearing' => $hearing->getId()]);
+        }
+
+        return $this->render('case/hearing/briefing/edit.html.twig', [
+            'translated_title' => $this->translator->trans('Edit briefing', [], 'case'),
+            'case' => $case,
+            'hearing' => $hearing,
+            'hearing_post_request' => $hearingPost,
+            'form' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @Route("/{case}/hearing/{hearing}/request/{hearingPost}/briefing/{briefing}/show", name="case_hearing_briefing_show")
+     */
+    public function show(CaseEntity $case, Hearing $hearing, HearingPostRequest $hearingPost, HearingBriefing $briefing): Response
+    {
+        $this->denyAccessUnlessGranted('edit', $case);
+
+        if ($hearing->getFinishedOn()) {
+            throw new HearingException();
+        }
+
+        return $this->render('case/hearing/briefing/show.html.twig', [
+            'translated_title' => $this->translator->trans('Create briefing', [], 'case'),
+            'case' => $case,
+            'briefing' => $briefing,
+        ]);
+    }
+
+    /**
+     * @Route("/{case}/hearing/{hearing}/request/{hearingPost}/briefing/cancel", name="case_hearing_briefing_cancel")
+     */
+    public function cancel(CaseEntity $case, Hearing $hearing, HearingPostRequest $hearingPost): Response
+    {
+        $hearingPost->setBriefExtraParties(HearingPostRequestType::BRIEFING_PARTIES_NO);
+        $this->entityManager->flush();
+
+        return $this->redirectToRoute('case_hearing_index', ['id' => $case->getId()]);
+    }
+
+    public function removeDocumentFromCase(CaseEntity $case, Document $document)
+    {
+        $relation = $this->relationRepository->findOneBy(['case' => $case, 'document' => $document]);
+
+        if (null !== $relation) {
+            $this->entityManager->remove($relation);
+            $this->entityManager->flush();
+        }
+
+        // Remove file
+        $this->documentUploader->deleteDocumentFile($document);
```

Comment:
Should we delete the document as well (not only the file)? Are we sure the the document is not used by other entities?

Reply:
The document was actually deleted after the call to this method, but it makes perfect to add this to the method.

Added 
``` 
$this->entityManager->remove($document);
```